ebos/damariswriter.hh

```diff
@@ -0,0 +1,384 @@
+                // which define sizes of the Damaris variables, per-rank and globally (over all ranks).
+                // Also sets the offsets to where a ranks array data sits within the global array. 
+                // This is usefull for HDF5 output and for defining distributed arrays in Dask.
+                this->setupDamarisWritingPars(simulator_.vanguard().grid().comm(), numElements_, elements_rank_offsets_);
+                
+                // sets data for non-time-varying variables MPI_RANK and GLOBAL_CELL_INDEX
+                this->SetGlobalIndexForDamaris() ; 
+                
+                // Currently by default we assume static grid (unchanging through the simulation)
+                // Set damarisUpdate_ to true if we want to update the geometry to sent to Damaris 
+                this->damarisUpdate_ = false; 
+            }
+            
+            
+            if (this->damarisOutputModule_->getPRESSURE_ptr() != nullptr) 
+            {
+                int64_t temp_int64_t[1];
+                temp_int64_t[0] = static_cast<int64_t>(this->elements_rank_offsets_[rank_]);
+                dam_err_ = damaris_set_position("PRESSURE", temp_int64_t);
+                if (dam_err_ != DAMARIS_OK && rank_ == 0) {
+                    OpmLog::error(fmt::format("ERORR: damariswriter::writeOutput()       : ( rank:{}) damaris_set_position(PRESSURE, ...), Damaris Error: {}  ",  rank_, damaris_error_string(dam_err_) ));
+                }
+                
+                dam_err_ = damaris_write("PRESSURE", (void*)this->damarisOutputModule_->getPRESSURE_ptr());
+                if (dam_err_ != DAMARIS_OK) {
+                   OpmLog::error(fmt::format("ERORR: damariswriter::writeOutput()       : ( rank:{}) damaris_write(PRESSURE, ...), Damaris Error: {}  ",  rank_, damaris_error_string(dam_err_) ));
+                }
+            
+            
+                dam_err_ =  damaris_end_iteration();
+                if (dam_err_ != DAMARIS_OK) {
+                    std::cerr << "ERROR rank =" << rank_ << " : damariswriter::writeOutput() : damaris_end_iteration()" 
+                    << ", Damaris error = " <<  damaris_error_string(dam_err_) << std::endl ;
+                }
+            }
+         } // end of ! isSubstep
+    }
+ 
+private:
+
+    int dam_err_ ;
+    int rank_  ;       
+    int nranks_ ;
+    int numElements_ ;  ///<  size of the unique vector elements
+    
+    Simulator& simulator_;
+    std::unique_ptr<EclOutputBlackOilModule<TypeTag>> damarisOutputModule_;
+    std::vector<unsigned long long> elements_rank_offsets_ ;
+    bool damarisUpdate_ = false;  ///< Whenever this is true writeOutput() will set up Damaris mesh information and offsets of model fields
+
+    static bool enableDamarisOutput_()
+    { 
+        return EWOMS_GET_PARAM(TypeTag, bool, EnableDamarisOutput); 
+    }
+
+    void SetGlobalIndexForDamaris () 
```

Comment:
In OPM, `lowerCamelCase()` should be used for functions, i.e. `SetGlobal...` -> `setGlobal...` .